test/compiler/e2e.ts

```diff
@@ -46,7 +48,11 @@ describe('E2E: Compiler integration', () => {
   });
 
   for (const configurationPath of configurations)  {
-    const configuration = loadConfig(configurationPath) as any;
+    const userConfig = await loadConfig(configurationPath);
+    const configuration = {
```

Comment:
I would propose to extract it to: 
```ts
const getConfig(configurationPath) : Configuration {
    const userConfig = await loadConfig(configurationPath);
    return {
      ...defaultConfig,
      ...userConfig
    };
}
```

Usage:
```js
const {name, targetPath, legacyOutput} = getConfig(configurationPath);
```

Reply:
points taken, but since the variable `configuration` will be used at https://github.com/EthWorks/Waffle/blob/ee1b9d54e807817f58aa511a95ff8a716e66653e/test/compiler/e2e.ts#L112
will simplify my last commit but keep configuration as a const var